action.yml

```
@@ -77,10 +77,149 @@ runs:
           }
         }
 
+        function Invoke-Download {
+          Param
+          (
+            [Parameter(Mandatory)]
+            [String] $URL,
+            [Alias("Destination")]
+            [String] $Path,
+            [Int] $Retries = 20
+          )
+
+          if (-not $Path) {
+            $InvalidCharacters = [IO.Path]::GetInvalidFileNameChars() -Join ''
+            $RE = "[{0}]" -f [RegEx]::Escape($InvalidCharacters)
+            $FileName = [IO.Path]::GetFileName($URL) -Replace $RE
+
+            if ([String]::IsNullOrEmpty($FileName)) {
+              $FileName = [System.IO.Path]::GetRandomFileName()
+            }
+            $Path = Join-Path -Path "${env:Temp}" -ChildPath $FileName
+          }
+
+          Write-Host "Downloading package from $URL to $Path..."
+
+          $Interval = 30
+          $StartTime = Get-Date
+          do {
+            try {
+              $AttemptStartTime = Get-Date
+              (New-Object System.Net.WebClient).DownloadFile($URL, $Path)
+              $AttemptDuration = [math]::Round(($(Get-Date) - $AttemptStartTime).TotalSeconds, 2)
+              Write-Host "Package downloaded in $AttemptDuration seconds"
+              break
+            } catch {
+              $AttemptDuration = [math]::Round(($(Get-Date) - $AttemptStartTime).TotalSeconds, 2)
+              Write-Warning "Package download failed after $AttemptDuration seconds"
+              Write-Warning $_.Exception.Message
+            }
+
+            if ($Retries -eq 1) {
+              $Duration = [math]::Round(($(Get-Date) - $StartTime).TotalSeconds, 2)
+              throw "Package download failed after $Duration seconds"
+            }
+
+            Write-Warning "Waiting $Interval seconds before retrying (retries remaining: $Retries)..."
+            Start-Sleep -Seconds $Interval
+          } while (--$Retries -gt 0)
+
+          return $Path
+        }
+
+        function Verify-Checksum {
+          Param
+          (
+            [Parameter(Mandatory = $true)]
+            [String]$Actual,
+            [Parameter(Mandatory = $true)]
+            [String]$Expected
```

Spaces to be consistent with the rest of the file.
```suggestion
            [Parameter(Mandatory)]
            [String] $Actual,
            [Parameter(Mandatory)]
            [String] $Expected
```